WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Notes Section/Customer Note/EditCustomerNoteViewModel.swift

```diff
@@ -14,19 +15,28 @@ final class EditCustomerNoteViewModel: EditCustomerNoteViewModelProtocol {
     ///
     @Published private(set) var navigationTrailingItem: EditCustomerNoteNavigationItem = .done(enabled: false)
 
-    /// Defaults to `nil`.
+    /// Indicates whether we must wait for the request before dismiss.
     ///
-    @Published var presentNotice: EditCustomerNoteNotice?
+    var shouldWaitForRequestIsFinishedToDismiss: Bool {
+        !areOptimisticUpdatesEnabled
+    }
 
-    /// Publisher accessor for `presentNotice`. Needed for the protocol conformance.
+    /// Presents an error notice in the tab bar context after the update operation fails.
     ///
-    var presentNoticePublisher: Published<EditCustomerNoteNotice?>.Publisher {
-        $presentNotice
-    }
+    private let noticePresenter: NoticePresenter
+
+    /// Presents an error notice in the current modal presentation context.
+    /// It's only needed when optimistic updates are not enabled.
+    ///
+    var modalNoticePresenter: NoticePresenter?
```

Comment:
As the comment says, we need this notice presenter because the modal is not dismissed when the optimistic updates are not enabled :/